tests/src/Kernel/Entity/GetBundleByBundleTest.php

```diff
@@ -0,0 +1,557 @@
+              'field_name' => 'group_audience_' . $group_audience_field_key,
+              'field_storage_config' => [
+                'settings' => [
+                  'target_type' => $group_entity_type_id,
+                ],
+              ],
+            ];
+
+            if (!empty($group_bundle_ids)) {
+              $settings['field_config'] = [
+                'settings' => [
+                  'handler_settings' => [
+                    'target_bundles' => array_combine($group_bundle_ids, $group_bundle_ids),
+                  ],
+                ],
+              ];
+            }
+            Og::createField(OgGroupAudienceHelper::DEFAULT_FIELD, $group_content_entity_type_id, $group_content_bundle_id, $settings);
+          }
+        }
+      }
+    }
+
+    // Test ::getGroupBundleIdsByGroupContentBundle().
+    foreach ($expected_group_by_group_content as $group_content_entity_type_id => $group_content_bundle_ids) {
+      foreach ($group_content_bundle_ids as $group_content_bundle_id => $expected_result) {
+        $this->assertEquals($expected_result, $this->groupManager->getGroupBundleIdsByGroupContentBundle($group_content_entity_type_id, $group_content_bundle_id));
+      }
+    }
+
+    // Test ::getGroupContentBundleIdsByGroupBundle().
+    foreach (['node', 'block_content'] as $group_entity_type_id) {
+      for ($i = 0; $i < 2; $i++) {
+        $group_bundle_id = 'group_' . $i;
+
+        // If the expected value is omitted, we expect an empty array.
+        $expected_result = !empty($expected_group_content_by_group[$group_entity_type_id][$group_bundle_id]) ? $expected_group_content_by_group[$group_entity_type_id][$group_bundle_id] : [];
+
+        $this->assertEquals($expected_result, $this->groupManager->getGroupContentBundleIdsByGroupBundle($group_entity_type_id, $group_bundle_id));
+      }
+    }
+  }
+
+  /**
+   * Provides test data for testGetBundleIdsByBundle().
+   *
+   * @return array
+   *   An array of test properties. Each property is an indexed array with the
+   *   following items:
+   *   - An array indicating the relationships between groups and group content
+   *     bundles that need to be set up in the test.
+   *   - An array containing the expected results for the call to
+   *     getGroupBundleIdsByGroupContentBundle().
+   *   - An array containing the expected results for the 4 calls to
+   *     getGroupContentBundleIdsByGroupBundle() that will be made in the test.
+   *     If an empty array is expected to be returned, this result is omitted.
+   */
+  public function getBundleIdsByBundleProvider() {
+    return [
+      // Test the simplest case: a single group content type that references a
+      // single group type.
+      [
+        // The first parameter sets up the relations between groups and group
+        // content.
+        [
+          // Creating group content of type 'node'.
+          'node' => [
+            // The first of which...
+            'group_content_0' => [
```

Comment:
ok, I think I start to follow - however what does `group_content_0` represent - a bundle name or an entity?